### PR TITLE
Basic framework to specify work that can be broken up.

### PR DESCRIPTION
**Pull request by ttung**
Three major components:

1. There is an abstract class `ChunkedTask` that specifies the work.  Calls to `run_one_unit()` will run one unit of work.  Calls to `get_state()` will serialize the state for subsequent execution.
2. There is an abstract class `ChunkedTaskRuntime` that specifies the environment for the work.  Calls to `get_remaining_time_in_millis()` should return the amount of time left in this current execution environment.
3. There is a class `ChunkedTaskRunner` that orchestrates the execution.

Add unit tests to test ChunkedTaskRunner.

### CODE REVIEW

File: tests/test_chunkedtask.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+        self.rounds_remaining -= 1
+
+        return self.rounds_remaining > 0
+
+
+class TestChunkedTaskRunner(unittest.TestCase):
+    def test_workload_resumes(self):
+        initial_state = (1, 1, 25)
+        expected_max_one_unit_runtime_millis = 10  # we know exactly how long we'll take.  we're so good at guessing!
```
Comment (kislyuk) [inline tests/test_chunkedtask.py]: :grin:

File: dss/events/chunkedtask/base.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import typing
+
+TaskStateType = typing.TypeVar('TaskStateType')
+RuntimeStateType = typing.TypeVar('RuntimeStateType')
+
+
+class Task(typing.Generic[TaskStateType]):
+    def run_one_unit(self) -> bool:
```
Comment (kislyuk) [inline dss/events/chunkedtask/base.py]: Consider using an iterator for ChunkedTask, instead of returning True or False.

Comment (ttung) [inline dss/events/chunkedtask/base.py]: In #323, I allow the task to return a value, so return values actually become significant.

**Comment by ttung** [comment]
> This looks good overall. I wonder why use a custom serialization scheme instead of base64-encoded pickle or json.dumps(object.dict)?

#325

**Pull request merged**